### PR TITLE
fix(buffer): the output document should be modifiable

### PR DESCRIPTION
**Pull request by xiyaowong**
Fixed errors related to "output" in #498, but not sure if all scenarios are covered. Actually, I haven’t encountered any other situations either.

### CODE REVIEW

**Comment by theol0403** [comment]
I think hardcoding output to be true is so that syntax highlighting is enabled for the output buffer.

**Comment by xiyaowong** [comment]
I don't think so. 

The output should not be treated as an external buffer, and the [code](https://github.com/vscode-neovim/vscode-neovim/blob/7094dc7208d7a4dca696bd131513a1a86b375fde/src/buffer_manager.ts#L228C1-L234C61) is also the same. Output is always created in vscode and always has `b:vscode_controlled`

File: src/buffer_manager.ts [hunk 1]
```diff
@@ -169,8 +169,11 @@ export class BufferManager implements Disposable, NeovimRedrawProcessable, Neovi
     }
 
     public isExternalTextDocument(textDoc: TextDocument): boolean {
+        // !Output should be modifiable, vscode treats it as a regular document.
```
Comment (justinmk) [inline src/buffer_manager.ts:172]:
```suggestion
        // !Output should be modifiable, vscode treats it as a regular document. #498
```

File: src/buffer_manager.ts [hunk 2]
```diff
@@ -169,8 +169,11 @@ export class BufferManager implements Disposable, NeovimRedrawProcessable, Neovi
...
+        // !When the option "modifiable" is set to false, nvim_buf_set_text will not work.
```
Comment (justinmk) [inline src/buffer_manager.ts:173]:
```suggestion
        // !When the option "modifiable" is set to false, nvim_buf_set_text will not work. #498
```

**Comment by xiyaowong** [comment]
@theol0403  It's best to merge this PR as soon as possible, as an error will affect other debugging.

**Pull request merged**